.github/workflows/trigger-ubuntu-win-build.yml

```diff
@@ -15,33 +15,74 @@ jobs:
   trigger-workflow:
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
       - name: Trigger Build workflow
         env:
           CI_PR_TOKEN: ${{ secrets.CI_PR_TOKEN }}
           PR_TITLE: ${{ github.event.pull_request.title }}
-          CI_PR: ${{ secrets.CI_REPO }}
+          CI_REPO: ${{ vars.CI_REPO }}
         run: |
-          $headers = @{
-            Authorization="Bearer $env:CI_PR_TOKEN"
-          }
-
-          # Private repository for builds
-          $apiRepoUrl = "https://api.github.com/repos/$env:CI_PR"
+          Import-Module ./helpers/GitHubApi.psm1
+          $gitHubApi = Get-GithubApi -Repository "${env:CI_REPO}" -AccessToken "${env:CI_PR_TOKEN}"
 
           $eventType = "trigger-${{ inputs.image_type }}-build"
-          $body = @{
-            event_type = $eventType;
-            client_payload = @{
-              pr_title = "$env:PR_TITLE"
-              custom_repo = "${{ github.event.pull_request.head.repo.full_name }}"
+          $clientPayload = @{
+              pr_title                = "${env:PR_TITLE}"
+              custom_repo             = "${{ github.event.pull_request.head.repo.full_name }}"
               custom_repo_commit_hash = "${{ github.event.pull_request.head.sha }}"
-            }
           }
 
-          $bodyString = $body | ConvertTo-Json
+          $gitHubApi.DispatchWorkflow($eventType, $clientPayload)
 
-          try {
-            Invoke-WebRequest -Uri "$apiRepoUrl/dispatches" -Method Post -Headers $headers -Body $bodyString | Out-Null
-          } catch {
-            throw "$($_.exception[0].message)"
-          }
+  wait-completion:
+    runs-on: ubuntu-latest
+    needs: trigger-workflow
+    outputs:
+      ci_workflow_run_id: ${{ steps.wait.outputs.ci_workflow_run_id }}
```

Comment:
nit: Consider resolving workflow run ID in the `trigger` job. The this job is solely responsible for awaiting the given workflow run and cancellation job would not need to depend on it (except for ordering reasons).

Ideally repo dispatch event would return workflow run id (which is a feature community has been longing for) and once it is supported in Actions we would simply switch to using it instead of resolving workflow id from the title.